src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -0,0 +1,253 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "network-commissioning.h"
+
+#include <cstring>
+
+#include <lib/support/CodeUtils.h>
+#include <lib/support/ReturnMacros.h>
+#include <lib/support/logging/CHIPLogging.h>
+#include <platform/CHIPDeviceLayer.h>
+#include <platform/ConnectivityManager.h>
+#include <transport/NetworkProvisioning.h>
+
+#if defined(CHIP_DEVICE_LAYER_TARGET)
+#define DEVICENETWORKPROVISIONING_HEADER <platform/CHIP_DEVICE_LAYER_TARGET/DeviceNetworkProvisioningDelegateImpl.h>
```

Comment:
why this odd #define followed by inclusion of the define?

can't we just include directly? if this is because macro expansion, please add a comment explaining.

Could this be resolved in some different way by using the build system?

Reply:
I would say we should remove DeviceNetworkProvisioningDelegateImpl put provide some platform neutral api in connectivity manager or some place else.
Currently we have other places (the rendezvous session) that is using DeviceNetworkProvisioningDelegateImpl, so I basically use this but I'd like to deprecate DeviceNetworkProvisioningDelegateImpl after we fully switched to network provisioning cluster.